### PR TITLE
Skip firmware reset on devices the operator doesn't control

### PR DESCRIPTION
**Pull request by SchSeba**
In case the user allocate vf or configure the mellanox card firmware and we don't have any policy in the operator we should not reset that configuration

### CODE REVIEW

File: pkg/plugins/mellanox/mellanox_plugin.go [hunk 1]
```diff
@@ -206,3 +211,21 @@ func (p *MellanoxPlugin) nicHasExternallyManagedPFs(nicPortsMap map[string]sriov
...
 }
+
+// nicNotConfiguredByOperator returns true if one of the ports(interface) of the NIC is not configured by operator
+func (p *MellanoxPlugin) nicNotConfiguredByOperator(nicPortsMap map[string]sriovnetworkv1.InterfaceExt) bool {
+	existPfStatusOnStore := false
+	for _, iface := range nicPortsMap {
+		_, exist, err := p.helpers.LoadPfsStatus(iface.PciAddress)
+		if err != nil {
+			log.Log.Error(err, "failed to load PF status from disk", "address", iface.PciAddress)
+			continue
```
Comment (adrianchiris) [inline pkg/plugins/mellanox/mellanox_plugin.go:222]: should we fail here ?

Comment (SchSeba) [reply pkg/plugins/mellanox/mellanox_plugin.go:222]: I wasn't sure I copy the other function :P 
https://github.com/k8snetworkplumbingwg/sriov-network-operator/pull/734/files/fa266810fd4c7bc0660cbcc750ac1df761c51038#diff-8f607793a0090ecce8926d771471ecce0f2514b77695e38bbd6bb70f8996f088R200

Comment (adrianchiris) [reply pkg/plugins/mellanox/mellanox_plugin.go:222]: fair enough :)

**Review by adrianchiris — APPROVED** [review]

File: pkg/plugins/mellanox/mellanox_plugin.go (hunk 1 above)
Comment (adrianchiris) [reply pkg/plugins/mellanox/mellanox_plugin.go:222]: perhaps we shoud consider failing in both places, LMK what you think i can submit a followup PR.

LoadPfStatus will fail if file exists but it failed reading from it or its corrupt and we werent able to unmarshall

File: pkg/plugins/mellanox/mellanox_plugin.go [hunk 2]
```diff
@@ -206,3 +211,21 @@ func (p *MellanoxPlugin) nicHasExternallyManagedPFs(nicPortsMap map[string]sriov
...
+		}
+		if exist {
+			log.Log.V(2).Info("PF configured by the operator", "interface", iface)
+			existPfStatusOnStore = true
+			break
```
Comment (zeeke) [inline pkg/plugins/mellanox/mellanox_plugin.go:227]:
```suggestion
			return false
```

can we get rid of the `existPfStatusOnStore` variable and early exit here?

**Review by zeeke — APPROVED** [review]
Only one minor comment from my side. LGTM

File: pkg/plugins/mellanox/mellanox_plugin.go (hunk 2 above)
Comment (SchSeba) [reply pkg/plugins/mellanox/mellanox_plugin.go]: right!

File: pkg/plugins/mellanox/mellanox_plugin.go (hunk 1 above)
Comment (SchSeba) [reply pkg/plugins/mellanox/mellanox_plugin.go:221]: sounds good I agree that should failed in both functions :) 

when you have time please open the PR I can review it

**Review by zeeke — APPROVED** [review]

**Pull request merged**